src/org/zaproxy/zap/extension/todo/TodoList.java

```diff
@@ -0,0 +1,148 @@
+import org.w3c.dom.NodeList;
+import org.zaproxy.zap.extension.tab.Tab;
+import org.apache.log4j.BasicConfigurator;
+import org.apache.log4j.Logger;
+
+public class TodoList extends AbstractPanel implements Tab {
+
+	private static final long serialVersionUID = 1L;
+	private JPanel panel;
+	JScrollPane thePane;
+	JScrollBar bar;
+	JPanel userDefined;
+	
+	public TodoList(){
+		this.setIcon(new ImageIcon(TodoList.class.getResource("/org/zaproxy/zap/extension/todo/resource/ui-check-boxes.png")));
+		this.setLayout(new BorderLayout());
+		panel = new JPanel();
+		panel.setLayout(new GridLayout(0,1));
+		//TODO : add user defined tasks
+		panel.add(new JLabel(Constant.messages.getString("todo.set.sheet.label")));
+		Logger logger = Logger.getLogger(TodoList.class);
+        	BasicConfigurator.configure();		
+		
+		try {
+			NodeList sections= parseXml();
+			for(int i =0 ;i<sections.getLength();i++){
+				Element section =  (Element) sections.item(i);
+				String name  = section.getElementsByTagName("name").item(0).getFirstChild().getNodeValue();
+				JLabel nodeLabel = new JLabel();
+				nodeLabel.setText(i+1 +". "+name);
+				panel.add(nodeLabel);
+				
+				Element tasks = (Element) section.getElementsByTagName("tasks").item(0);
+				NodeList taskList = tasks.getElementsByTagName("task");
+				
+				for(int j=0;j<taskList.getLength();j++){
+					String taskName  = taskList.item(j).getFirstChild().getNodeValue();
+					ListItem item = new ListItem(taskName);
+					Component handle = panel.add(item);
+					item.handle = handle;
+				}
+			}
+			
+		} catch (Exception e) {
+			logger.info("Error occurred while parsing cheatsheet");
+		}
+		
+		thePane = new JScrollPane(panel);
+		bar = thePane.getVerticalScrollBar();
+		this.add(thePane);
+		
+	}
+
+	class ListItem extends JComponent{
+		private static final long serialVersionUID = 1L;
+		JCheckBox checkbox;
+		JLabel label;
+		JButton removeBtn;
+		Component handle;
+		
+		public ListItem(String label) {
+			
+			setLayout(new FlowLayout(FlowLayout.LEFT));
+			this.label = new JLabel();
+			this.label.setText(label);
+			
+			this.checkbox = new JCheckBox();
+			
+			this.removeBtn = new JButton();
+			removeBtn.setText(Constant.messages.getString("todo.remove.btn.label"));
+			removeBtn.addActionListener(new ActionListener() {
+				
+				@Override
+				public void actionPerformed(ActionEvent e) {
+					panel.remove(handle);
+				    panel.repaint();
+				    crazyFix(); //clicking on remove btn does not immediately rerender the view.so use fix
+				}
+			});
+			add(checkbox);
+	        add(this.label);
+	        add(removeBtn);
+	        
+		}
+	}
+	
+	public NodeList parseXml() throws Exception{
+		
+		DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+		DocumentBuilder db = dbf.newDocumentBuilder();
+		File xmlCheatSheet = new File((TodoList.class.getResource("/org/zaproxy/zap/extension/todo/resource/OWASP_Web_Application_Testing_Cheat_Sheet.xml")).toURI());
```

Comment:
- There's a typo in `resources`.
 - Might need to use `getResourceAsStream` (instead of `getResource`) for parsing, the file is now bundled in the add-on.